### PR TITLE
refactor: move brc20 validation logic into core

### PR DESCRIPTION
**Pull request by teebszet**
# 🔘 PR Type
- [x] Refactoring (no functional changes, no api changes)

# 📜 Background
move brc20 validation logic into core

# 🔄 Changes

Does this PR introduce a breaking change?

- [ ] Yes, Incompatible API changes
- [x] No, Adds functionality (backwards compatible)
- [ ] No, Bug fixes (backwards compatible)

# ✅ Review checklist

Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: utils/brc20.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+const isNumber = (value: string) => !Number.isNaN(Number(value));
+
+export type Brc20Definition = {
+  op: 'deploy' | 'mint' | 'transfer';
+  tick: string;
+  value: string;
+};
+
+export const getBrc20Details = (content: string, contentType: string): undefined | Brc20Definition => {
```
Comment (fedeerbes) [inline utils/brc20.ts:16]: quick question. do we want to write a few UT on this?

Comment (teebszet) [reply utils/brc20.ts:16]: absolutely

Comment (teebszet) [reply utils/brc20.ts:16]: done @fedeerbes. thanks for encouraging 😄

**Review by fedeerbes — APPROVED** [review]
LGTM 🙌🏼

**Pull request merged**